list.go

```diff
@@ -0,0 +1,31 @@
+package ctxvfs
+
+import "context"
+
+// ListAllFileser is type that implements ListAllFiles
+type ListAllFileser interface {
+	// ListAllFiles returns a slice of all file paths on a VFS. This is
+	// only files (excludes directories). This interface exists for
+	// implementors to provide a faster way to list all files than walking
+	// the file tree with ReadDir/etc.
+	ListAllFiles(ctx context.Context) ([]string, error)
+}
+
+// ListAllFiles returns a slice of all file paths (excludes directories).
+func ListAllFiles(ctx context.Context, fs FileSystem) ([]string, error) {
```

Comment:
You are suggesting a VFS wrapper, which makes me go ugh. Can we rather not rely on convention/documentation to signal the slower path is possible? In production we would notice this in our traces/metrics if we did something silly.

At the same time I am happy to actually just abandon this PR and use your original suggestion internally in our codebase.